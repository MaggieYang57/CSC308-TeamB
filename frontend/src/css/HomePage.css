.top-banner{
    display: flex;
    justify-content: center;
    padding-top: 5vw;
    padding-bottom: 5vw;
    background-color: #80cbc4;
}

.banner-info{
    display: flex;
    text-align: left;
    align-items: center;
    justify-content: center;
    background-color: white;
    margin-right: 8vw;
    margin-left: 8vw;
    padding: 50px;
    background-clip: padding-box;
    filter: drop-shadow(5px 10px 5px grey);
    overflow: hidden;
}
.banner-info img{
    justify-content: left;
    border-radius: 50px;
    margin-left: 1vw;
}
.banner-info h1{
    font-family: 'Mulish', sans-serif;
    margin-left: 2vw;
    font-size: 30px;
    font-weight: 800;
    color:#265b5f;
}
.banner-info p{
    margin-left: 2vw;
    font-family: 'Quicksand', sans-serif;
    font-size: 15px;
}
.banner-info button{
    margin-left: 15vw;
    padding: 15px 20px;
    border-radius: 20px;
    border: none;
    font-family: Arial, Helvetica, sans-serif;
    font-weight: 600;
    font-size: 15px;
    letter-spacing: 1px;
    text-decoration: none;
    color:white;
    background-color: #ff8c69;
    -webkit-transition-duration: 0.2s;
    transition-duration: 0.2s;
}
.banner-info button:hover, .banner-info button-fade:focus, .banner-info button-fade:active {
    text-decoration: none;
    background-color: #2098D1;
    color: white;
}

#featured-title
{
    text-align: left;
    margin-top: 6vw;
    color:#265b5f;
}

/*********************columns***************************/
.row {
    width: 100%;
    margin-bottom: 10vw;
    
}
.row:after{
    content: "";
    display: table;
    clear: both;
}
.row img{
    display: flex;
    justify-content: left;
    margin-top: 30px;
    margin-left: 0px;
    border-radius: 30px;
}
#column1{
    margin-left:100px;
    float: left;
    width: 28%;
    padding-top: 2vw;
    padding-bottom: 2vw;
    background-color:white;
    overflow: hidden;
    border-top-style: solid;
    border-bottom-style: solid;
    border-width: 1px;
    border-color: gray;
}
#column2{
    float: left;
    width: 28%;
    padding-top: 2vw;
    padding-bottom: 2vw;
    background-color:white;
    overflow: hidden;
    border-top-style: solid;
    border-bottom-style: solid;
    border-width: 1px;
    border-color: gray;
}
#column3{
    float: left;
    width: 28%;
    padding-top: 2vw;
    padding-bottom: 2vw;
    background-color:white;
    overflow: hidden;
    border-top-style: solid;
    border-bottom-style: solid;
    border-width: 1px;
    border-color: gray;
}
.info{
    text-align: left;
    margin-top: -140px;
    margin-left: 190px;
    margin-right: -10px;
}

.info h1{
    text-align: left;
    font-family: Arial, Helvetica, sans-serif;
    font-weight: 700;
    font-size: 25px;
    color:rgb(0, 0, 0);
}
.info h2{
    margin-top: -20px;
    text-align: left;
    font-family: Arial, Helvetica, sans-serif;
    letter-spacing: 2px;
    font-weight: 700;
    font-size: 20px;
    color:rgb(0, 0, 0);
}
.info p{
    text-align: left;
    font-family: 'Quicksand', sans-serif;
    font-size: 15px;
    color: black;
}
