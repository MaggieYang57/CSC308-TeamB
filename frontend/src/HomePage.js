/* eslint-disable react/jsx-key */
import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Card, Row, Col } from "react-bootstrap";
import "./css/HomePage.css";
import "bootstrap/dist/css/bootstrap.min.css";

function HomePage(props) {
  return (
    <div className="home">
      <div className="top-banner">
        <div className="banner-info">
          <img
            src="https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg"
            height="270"
          />
          <div>
            <b
              className="text text-center "
              style={{ fontSize: 35, color: "#2C6674", marginLeft: "2vw" }}
            >
              Want to get out and explore?
            </b>
            <p style={{ marginTop: "1vw" }}>
              There are so many great hikes to go on in SLO county and the city
              of San Luis Obispo itself. With SLO HIKES, the perfect hike for
              you is only a few clicks away. Here you will find relevant reviews
              and useful filtering functionality to make your experience a
              breeze. Help us improve the hiking community in SLO County and
              make some more memories in the beautiful outdoors!
            </p>
            <Link to="/hikeFinder">
              <button href="hikefinder">FIND A HIKE NOW</button>
            </Link>
          </div>
        </div>
      </div>
      <div className="featured-hikes">
        <div id="featured-title">
          <b
            className="text text-left"
            style={{ fontSize: 35, color: "#2C6674", marginLeft: "7vw" }}
          >
            Most Popular Hikes
            <hr />
          </b>
        </div>
        <div
          className="card-deck"
          style={{ marginLeft: "1.5vw", marginRight: "1.5vw" }}
        >
          <Col>
            <Row className="mb-5" style={{ margin: 20 }}>
              {props.hikeList.slice(0, 1).map((hike) => (
                <Col>
                  <Card.Body>
                    <Card.Title>
                      <Card.Img
                        variant="top"
                        src={hike.imagesrc}
                        fluid="true"
                      />
                      <Link to={"/hike/" + hike._id}>
                        <p
                          className=" h5 card-text text-left text-primary font-weight-bold"
                          style={{ marginTop: "2vw" }}
                        >
                          {hike.title}
                        </p>
                      </Link>
                    </Card.Title>
                    <Card.Text>
                      <p
                        className="card-text text-left"
                        style={{ fontWeight: "bold" }}
                      >
                        {hike.location}
                      </p>
                    </Card.Text>

                    <Card.Text>
                      <p className="card-text text-left">
                        {hike.description.substring(0, 150) + "..."}
                      </p>
                    </Card.Text>
                  </Card.Body>
                </Col>
              ))}
              {props.hikeList.slice(1, 2).map((hike) => (
                <Col>
                  <Card.Body>
                    <Card.Title>
                      <Card.Img
                        variant="top"
                        src={hike.imagesrc}
                        fluid="true"
                      />
                      <Link to={"/hike/" + hike._id}>
                        <p
                          className=" h5 card-text text-left text-primary font-weight-bold"
                          style={{ marginTop: "2vw" }}
                        >
                          {hike.title}
                        </p>
                      </Link>
                    </Card.Title>
                    <Card.Text>
                      <p
                        className="card-text text-left"
                        style={{ fontWeight: "bold" }}
                      >
                        {hike.location}
                      </p>
                    </Card.Text>

                    <Card.Text>
                      <p className="card-text text-left">
                        {hike.description.substring(0, 150) + "..."}
                      </p>
                    </Card.Text>
                  </Card.Body>
                </Col>
              ))}
              {props.hikeList.slice(2, 3).map((hike) => (
                <Col>
                  <Card.Body>
                    <Card.Title>
                      <Card.Img
                        variant="top"
                        src={hike.imagesrc}
                        fluid="true"
                      />
                      <Link to={"/hike/" + hike._id}>
                        <p
                          className=" h5 card-text text-left text-primary font-weight-bold"
                          style={{ marginTop: "2vw" }}
                        >
                          {hike.title}
                        </p>
                      </Link>
                    </Card.Title>
                    <Card.Text>
                      <p
                        className="card-text text-left"
                        style={{ fontWeight: "bold" }}
                      >
                        {hike.location}
                      </p>
                    </Card.Text>

                    <Card.Text>
                      <p className="card-text text-left">
                        {hike.description.substring(0, 150) + "..."}
                      </p>
                    </Card.Text>
                  </Card.Body>
                </Col>
              ))}
            </Row>
          </Col>
        </div>
      </div>
    </div>
  );
}

HomePage.propTypes = {
  hikeList: PropTypes.object,
};
export default HomePage;
