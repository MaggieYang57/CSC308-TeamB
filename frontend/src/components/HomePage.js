import React from 'react';
import { Link } from "react-router-dom";
import '../css/HomePage.css';

class HomePage extends React.Component {
  render() {
    return (
        <div className="home">
            <div className= "top-banner">
                <div className = "banner-info">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="270" />
                    <div>
                        <h1>Want to get out and explore?</h1>
                        <p>This body of text should give a little info on our site and what users can do here. Explain breifly how we can help them, why they want to stay and use our site, maybe a little bit about what our mission is. But not too much text, need to keep it light. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras sodales, diam eget tristique ultrices, orci massa consequat sapien, nec egestas sem tellus et nisi. </p>
                        <Link to="/hikeFinder"><button href="hikefinder">FIND A HIKE NOW</button></Link>
                    </div>
                </div>
            </div>
            <div className = "featured-hikes">
                <h1 id="featured-title">Most Popular Hikes</h1>
                <div className = "row">
                    <div id = "column1">
                        <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                    <div id = "column2">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                    <div id = "column3">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
  }
}
export default HomePage;