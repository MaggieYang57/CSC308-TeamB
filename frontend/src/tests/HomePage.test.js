import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import HomePage from '../HomePage';

it('renders correctly', () => {
  const hikes  = [
    {
      '_id': 1, 
      'rating': [5, 3, 4], 
      'title':'Hike 1', 
      'location': 'SLO', 
      'description': 'great hike', 
      'imagesrc': 'test' 
    },
    {
      '_id': 2, 
      'rating': [5, 3, 4], 
      'title':'Hike 2', 
      'location': 'BILL', 
      'description': 'Amazing getaway to the green beyond.', 
      'imagesrc': 'hills' 
    },
    {
      '_id': 3, 
      'rating': [5, 3, 4], 
      'title':'Hike 3', 
      'location': 'Avila', 
      'description': 'Great hike for destressing', 
      'imagesrc': 'Tall hike' 
    }
  ]
  const tree = renderer
    .create(<MemoryRouter> <HomePage hikeList={hikes} /> </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});