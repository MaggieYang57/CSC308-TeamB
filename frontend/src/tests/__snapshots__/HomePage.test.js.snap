// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <div
    className="home"
  >
    <div
      className="top-banner"
    >
      <div
        className="banner-info"
      >
        <img
          height="270"
          src="https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg"
        />
        <div>
          <b
            className="text text-center "
            style={
              Object {
                "color": "#2C6674",
                "fontSize": 35,
                "marginLeft": "2vw",
              }
            }
          >
            Want to get out and explore?
          </b>
          <p
            style={
              Object {
                "marginTop": "1vw",
              }
            }
          >
            There are so many great hikes to go on in SLO county and the city of San Luis Obispo itself. With SLO HIKES, the perfect hike for you is only a few clicks away. Here you will find relevant reviews and useful filtering functionality to make your experience a breeze. Help us improve the hiking community in SLO County and make some more memories in the beautiful outdoors!
          </p>
          <a
            href="/hikeFinder"
            onClick={[Function]}
          >
            <button
              href="hikefinder"
            >
              FIND A HIKE NOW
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="featured-hikes"
    >
      <div
        id="featured-title"
      >
        <b
          className="text text-left"
          style={
            Object {
              "color": "#2C6674",
              "fontSize": 35,
              "marginLeft": "7vw",
            }
          }
        >
          Most Popular Hikes
          <hr />
        </b>
      </div>
      <div
        className="card-deck"
        style={
          Object {
            "marginLeft": "1.5vw",
            "marginRight": "1.5vw",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="mb-5 row"
            style={
              Object {
                "margin": 20,
              }
            }
          >
            <div
              className="col"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  <img
                    className="card-img-top"
                    fluid="true"
                    src="test"
                  />
                  <a
                    href="/hike/1"
                    onClick={[Function]}
                  >
                    <p
                      className=" h5 card-text text-left text-primary font-weight-bold"
                      style={
                        Object {
                          "marginTop": "2vw",
                        }
                      }
                    >
                      Hike 1
                    </p>
                  </a>
                </div>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    SLO
                  </p>
                </p>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                  >
                    great hike...
                  </p>
                </p>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  <img
                    className="card-img-top"
                    fluid="true"
                    src="hills"
                  />
                  <a
                    href="/hike/2"
                    onClick={[Function]}
                  >
                    <p
                      className=" h5 card-text text-left text-primary font-weight-bold"
                      style={
                        Object {
                          "marginTop": "2vw",
                        }
                      }
                    >
                      Hike 2
                    </p>
                  </a>
                </div>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    BILL
                  </p>
                </p>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                  >
                    Amazing getaway to the green beyond....
                  </p>
                </p>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  <img
                    className="card-img-top"
                    fluid="true"
                    src="Tall hike"
                  />
                  <a
                    href="/hike/3"
                    onClick={[Function]}
                  >
                    <p
                      className=" h5 card-text text-left text-primary font-weight-bold"
                      style={
                        Object {
                          "marginTop": "2vw",
                        }
                      }
                    >
                      Hike 3
                    </p>
                  </a>
                </div>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Avila
                  </p>
                </p>
                <p
                  className="card-text"
                >
                  <p
                    className="card-text text-left"
                  >
                    Great hike for destressing...
                  </p>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
